index_alias_impl.go

```diff
@@ -205,14 +205,34 @@ func (i *indexAliasImpl) SearchInContext(ctx context.Context, req *SearchRequest
 	// only perform presearch if
 	//  - the request does not already have preSearchData
 	//  - the request requires presearch
+	var preSearchDuration time.Duration
 	if req.PreSearchData == nil && PreSearchRequired(ctx, req) {
-		preSearchData, err = PreSearch(ctx, req, i.indexes...)
+		searchStart := time.Now()
+		preSearchResult, err := PreSearch(ctx, req, i.indexes...)
 		if err != nil {
 			return nil, err
 		}
+		// check if the presearch result has any errors and if so
+		// return the search result as is without executing the query
+		// so that the errors are not lost
+		if preSearchResult.Status.Failed > 0 {
+			return preSearchResult, nil
+		} else {
+			// if there are no errors, then use the presearch data
+			// to execute the query
+			preSearchData, err = mergePreSearchData(req, preSearchResult, len(i.indexes))
+			if err != nil {
+				return nil, err
+			}
+		}
```

Comment:
All this code (preSearchResult.status check and mergePreSearchData) was better positioned where it was. Let's move this back to where it was and just do the preSearchDuration calculation here.

Reply:
I understand why you're doing this though - so this is ok.